code/game/objects/objs.dm

```diff
@@ -18,6 +18,22 @@
 	var/holographic = 0 //if the obj is a holographic object spawned by the holodeck
 	var/tmp/directional_offset ///JSON list of directions to x,y offsets to be applied to the object depending on its direction EX: {'NORTH':{'x':12,'y':5}, 'EAST':{'x':10,'y':50}}
 
+	///The current health of the obj. Leave to null, unless you want the object to start at a different health than max_health.
+	var/health
+	///The maximum health that the object can have. If set to ITEM_HEALTH_NO_DAMAGE, the object won't take any damage.
+	var/max_health = ITEM_HEALTH_NO_DAMAGE
+
+/obj/Initialize(mapload)
+	//Health should be set to max_health only if it's null.
+	if(isnull(health))
```

Comment:
Yeah, but that's voluntary. Some things do not start at full health, like vines, blobs, and a few other things.  So you don't want to automatically set those to max_health. And if you'd do the same check inside a material update proc, you'd have to have some way of knowing if you can mess with the health value or not since material could change at runtime as well as being set during init. 
It also makes my life much easier downstream when loading from save something with a saved health.

I'm not really thrilled about making a get_max_health proc that would return something that may change over time. Because that means having to figure out when that value change and update health accordingly. Like material durability could easily just be calculated into max_health as it is. And it would be easier to keep a coherent internal state between changes to health values.

Currently, only at the /obj level, I'm gating health and max_health changes behind set_max_health and set_health. And set_max_health has a flag for telling how to handle updating the current health so it's a bit more flexible (for now those flags are: adding the difference to health, leaving health alone, reseting health to max_health, or just clamping health to max_health).

Also, max_health is going to be accessed a lot, and using a proc might have an overhead. Especially for things that call health procs in a loop, or through subsystems.
And, not all atoms really have a concept of health. And usually you can easily replace most things that would needs to access get_max_health values with something getting a percentage of damage/health, which furthers makes it easier to abstract things out.